adapters/beachfront/beachfronttest/supplemental/banner-and-adm-video-by-explicit-204-response-on-banner.json

```diff
@@ -0,0 +1,147 @@
+              "w": 300,
+              "h": 250
+            }
+          ]
+        }
+      }
+    ],
+    "site": {
+      "page": "https://some.domain.us/some/page.html"
+    },
+    "device":{
+      "ip":"255.255.255.255"
+    }
+  },
+
+  "httpCalls": [
+    {
+      "expectedRequest": {
+        "uri": "https://qa.beachrtb.com/prebid_display",
+        "body": {
+          "slots": [
+            {
+              "slot": "mix1",
+              "id": "bannerAppId1",
+              "bidfloor": 0.41,
+              "sizes": [
+                {
+                  "w": 300,
+                  "h": 250
+                }
+              ]
+            }
+          ],
+          "domain": "some.domain.us",
+          "page": "https://some.domain.us/some/page.html",
+          "real204": true,
+          "referrer": "",
+          "search": "",
+          "secure": 1,
+          "deviceOs": "",
+          "deviceModel": "",
+          "isMobile": 0,
+          "ua": "",
+          "ip": "255.255.255.255",
+          "dnt": 0,
+          "user": {},
+          "adapterName": "BF_PREBID_S2S",
+          "adapterVersion": "0.9.2",
+          "requestId": "banner-and-video"
+        }
+      },
+      "mockResponse": {
+        "status": 204,
+        "body": ""
+      }
+    },
+    {
+      "expectedRequest": {
+        "uri": "https://qa.beachrtb.com/bid.json?exchange_id=videoAppId1",
+        "body": {
+          "id": "banner-and-video",
+          "imp": [
+            {
+              "video": {
+                "w": 300,
+                "h": 250,
+                "mimes": [
+                  "video/mp4"
+                ]
+              },
+              "bidfloor": 0.41,
+              "secure": 1,
+              "id": "mix1"
+            }
+          ],
+          "site": {
+            "domain": "some.domain.us",
+            "page": "https://some.domain.us/some/page.html"
+          },
+          "device": {
+            "devicetype": 2,
+            "ip": "255.255.255.255"
+          },
+          "cur": [
+            "USD"
+          ]
+        }
+      },
+      "mockResponse": {
+        "status": 200,
+        "body": {
+          "id": "61b87329-8790-47b7-90dd-c53ae7ce1723",
+          "seatbid": [
+            {
+              "bid": [
+                {
+                  "id": "5d839458f73decdc1572b7f6",
+                  "impid": "mix1",
+                  "price": 20,
+                  "adm": "<VAST version=\"2.0\"><Ad><Wrapper>http://example.com/vast.xml</Wrapper></Ad></VAST>",
+                  "adid": "1088",
+                  "adomain": [
+                    "beachfront.io"
+                  ],
+                  "cid": "277",
+                  "crid": "532",
+                  "w": 300,
+                  "h": 250
+                }
+              ],
+              "seat": "bfb-io-s1"
+            }
+          ],
+          "bidid": "5d839458f73decdc1572b7f6",
+          "cur": "USD"
+        }
+      }
+    }
```

Comment:
@mansinahar - This gets into some deeper questions and concerns that I've had for a while and I think I need to get to the bottom of them.

Many of the test cases do not behave like the real requests. Specifically, the "expectedBidResponses" works if I break it up into multiple "bids" objects, each with an array of "bid" objects (see banner-and-adm-video.json), wrapped in an object, but for the same request sent to a running server, I get back a single "seatbid" with an array of "bid" objects, wrapped in an object. I'll send you some gists in a bit.

For the test above, if I send an identical request (substituting "beachfront" for "bidder", of course) to a running server with a bad appid for the banner, therefore getting a 204, I still get back a video response, and it works just fine. However, the only "expectedBidResponses" that I've been able to make work in the test is an empty array, which does not reflect what bid responses I actually expect. If I include anything like what I actually expect, like just the video bid, or like the banner-and-adm-video.json, but with an empty array for the banner bid, I get a runtime error.

So, I suspect that this is happening because the beachfront bidder uses multiple endpoints. I've gotten a better understanding of how the validate.sh tests work through my last couple PRs (I think). I've also dug through that code thinking maybe this should be called an issue with the test system. I'm now thinking that's the wrong  thinking. I think that the prebid-server system assumes a bidder calls one endpoint, and I've hacked around that. **I think that the beachfront bidder should be broken up into 3 separate bidders.** Does that make sense?

**As far as you know - is anyone else using multiple endpoints in a single bidder?**

Also, I think that this is related to the crazy mess that I had with trying to incorporate multiple impressions for the AdM video requests. That was working fine when I did mixed requests to a running server, but failed on tests with runtime errors.

Reply:
Here are some requests and responses from running the test json to a running instance:

banner and adm gist:
https://gist.github.com/muncha/f18360946e6ddf8d6a3205adf8dc8913

banner(204) and adm gist:
https://gist.github.com/muncha/74b4bd025fa07ce12b1e7e344db5aba5

Reply:
Done. 

The change that I made in adapters/adapterstest/test_json.go got this test to work, and makes sense to me. Without that the test throws a runtime [error](https://gist.githubusercontent.com/muncha/74b4bd025fa07ce12b1e7e344db5aba5/raw/aa1e4967ff33b8c6ff01c4afb970cd38726890c7/runtime_error.txt).

Reply:
As I mentioned in some now deleted comments where I babbled on quite a bit, sending the same request to a running instance does not raise a runtime error and works just fine.

https://gist.github.com/muncha/74b4bd025fa07ce12b1e7e344db5aba5